engine/apps/schedules/tests/test_custom_on_call_shift.py

```diff
@@ -1826,3 +1831,69 @@ def test_refresh_schedule(make_organization_and_user, make_schedule, make_on_cal
     assert mock_refresh_final.apply_async.called
     assert schedule.cached_ical_file_primary is not None
     assert schedule.cached_ical_file_overrides is not None
+
+
+@pytest.mark.parametrize(
+    "users_per_group, shift_start, day_mask, total_days, frequency, interval, expected_result", CUSTOM_SHIFT_TEST_CASES
+)
+@pytest.mark.django_db
+def test_ical_shift_generation(
+    make_organization,
+    make_user_for_organization,
+    make_schedule,
+    make_on_call_shift,
+    users_per_group,
+    shift_start,
+    day_mask,
+    total_days,
+    frequency,
+    interval,
+    expected_result,
+):
+    organization = make_organization()
+    schedule = make_schedule(
+        organization,
+        schedule_class=OnCallScheduleWeb,
+        name="test_web_schedule",
+    )
+    total_users = sum(users_per_group)
+    users = [make_user_for_organization(organization, username=chr(i + 64)) for i in range(1, total_users + 1)]
+
+    start = datetime.strptime(shift_start, "%Y-%m-%d").replace(tzinfo=ZoneInfo("UTC"))
+    data = {
+        "start": start,
+        "rotation_start": start,
+        "until": start + timezone.timedelta(days=total_days),
+        "duration": timezone.timedelta(hours=12),
+        "frequency": frequency,
+        "by_day": day_mask,
+        "schedule": schedule,
+        "interval": interval,
+        "priority_level": 1,
+        "week_start": CustomOnCallShift.MONDAY,
```

Comment:
Does this get set to Monday in all cases? It seems to be set in the serializer to Monday when it is None.  DB default is Sunday but without this being set to Monday not all of these test cases pass.